logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt

```diff
@@ -57,10 +59,15 @@ class MessageSenderImpl(
             attemptToSend(conversationId, message)
                 .flatMap { messageRemoteTime ->
                     messageRepository.updateMessageDate(conversationId, message.id, messageRemoteTime)
-                    messageRepository.markMessageAsSent(conversationId, message.id)
-                    val millisDiff = message.date.toInstant().until(messageRemoteTime.toInstant(), DateTimeUnit.MILLISECOND)
-                    // this should make sure that pending messages are ordered correctly after one of them is sent
-                    messageRepository.updatePendingMessagesAddMillisToDate(conversationId, millisDiff)
+                        .flatMap {
+                            messageRepository.updateMessageStatus(MessageEntity.Status.SENT, conversationId, message.id)
+                        }.flatMap {
+                            // this should make sure that pending messages are ordered correctly after one of them is sent
+                            messageRepository.updatePendingMessagesAddMillisToDate(
+                                conversationId,
+                                timeParser.calculateMillisDifference(message.date, messageRemoteTime)
```

Comment:
this line 
`val millisDiff = message.date.toInstant().until(messageRemoteTime.toInstant(), DateTimeUnit.MILLISECOND)`

was requiring to pass a valid date to messageRepository.getMessageById when mocking the messageRepository for unit tests, also it required to some other function to return a valid messageRemoteTime format. 

I solved it by creating an interface that we can mock, this method :
val millisDiff = message.date.toInstant().until(messageRemoteTime.toInstant(), DateTimeUnit.MILLISECOND)

could throw an error, when it does, the app crashes, but we both discussed with Michal that it should never happen we think